tests/test_runner.py

```diff
@@ -359,8 +359,8 @@ def test_recipe_set_to_job_whst(self):
 
         result = self.myrunner._BeakerRunner__recipe_set_to_job(xml_parsed)
 
-        # check that <hostname op="!=" value="hst1"/> was removed
-        self.assertEqual(result.findall('.//hostname'), [])
+        # check that <hostname op="!=" value="hst1"/> wasn't removed
+        self.assertEqual(len(result.findall('.//hostname')), 1)
```

Comment:
Why check only the length and not the actual content? Does `findall` return an annoying object that is hard to compare or something?

Reply:
It's an element object that needs to be transformed to string in order to be easily compared. I can do that if you think it's useful

Reply:
I know this is merged, but... I think it's ok as-is, I just wondered why it was originally a comparison between the lists, and then just their lengths. However, as a possible follow-up, there are two cases that come to mind that I wonder if they are worth testing:

1. Non-ASCII in the tests? I guess it won't be a problem for hostnames, but I wonder if we have any reason to think that non-ASCII characters could be a problem, in which case we should add a case for that.
1. If there is more than one result, does the order matter? Could the order become a problem? If so, test for it.

Reply:
1. By non-ascii, are you talking about the decodes? Those are normal for python3 as it defaults to bytes. If you mean actual unexpected unicode characters, I think we're fine here as we control the passed values.

2. No, order shouldn't matter.

Reply:
I'm not sure what I meant with the first point, haha. I guess if we're using well-tested libraries for all this, it shouldn't be a problem. The producer of that XML is also using a library, I assume? No `printf`-like things or string concatenating or anything, I hope... So just ignore my comment.